core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,179 @@
+    this.desc = storyModel.getTitle();
+    this.imgColor = storyModel.getThumbnailBgColor();
+    this.imgUrl = this.getStorySummaryThumbnailUrl({
+      filename: storyModel.getThumbnailFilename(),
+      id: storyModel.getId(),
+    });
+
+    const nextStory =
+      completedStories === storyModel.getAllNodes().length
+        ? completedStories - 1
+        : completedStories;
+
+    const lessonArgs = {
+      classroom: storyModel.getClassroomUrlFragment(),
+      topicFragment: storyModel.getTopicUrlFragment(),
+      currentStory: storyModel.getAllNodes()[nextStory],
+      story: storyModel.getUrlFragment(),
+    };
+    this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+    this.title = `Chapter ${nextStory + 1}: ${storyModel.getNodeTitles()[nextStory]}`;
+    this.progress = Math.floor(
+      (completedStories / storyModel.getNodeTitles().length) * 100
+    );
+    this.lessonTopic = this.topic;
+  }
+
+  setCollectionSummary(collectionModel: CollectionSummary): void {
+    this.desc = collectionModel.objective;
+    this.imgColor = collectionModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      collectionModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = collectionModel.title;
+    this.lessonUrl = `/collection/${collectionModel.id}`;
+    this.lessonTopic = 'Collections';
+  }
+
+  setExplorationSummary(explorationModel: LearnerExplorationSummary): void {
+    this.desc = explorationModel.objective;
+    this.imgColor = explorationModel.thumbnailBgColor;
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      explorationModel.thumbnailIconUrl
+    );
+    this.progress = 0;
+    this.title = explorationModel.title;
+    this.lessonUrl = `/explore/${explorationModel.id}`;
+    this.lessonTopic = 'Community Lessons';
+  }
+
+  getStorySummaryThumbnailUrl({filename, id}: ThumbnailUrl): string {
+    if (!filename) {
+      return this.urlInterpolationService.getStaticImageUrl(
+        '/subjects/Lightbulb.svg'
+      );
+    }
+    return this.assetsBackendApiService.getThumbnailUrlForPreview(
+      AppConstants.ENTITY_TYPE.STORY,
+      id,
+      filename
+    );
+  }
+
+  getStorySummaryLessonUrl({
+    classroom,
+    topicFragment,
+    story,
+    currentStory,
+  }: LessonUrl): string {
+    return classroom === undefined || topicFragment === undefined
+      ? '#'
+      : `/explore/${currentStory.getExplorationId()}?` +
+          Object.entries({
+            topic_url_fragment: topicFragment,
+            classroom_url_fragment: classroom,
+            story_url_fragment: story,
+            node_id: currentStory.getId(),
+          })
+            .map(([key, value]) => `${key}=${value}`)
+            .join('&');
+  }
+
+  handleImageError(): void {
+    this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+      '/subjects/Lightbulb.svg'
+    );
+  }
+
+  setButtonText(): string {
```

Comment:
This name could be better. This function doesn't set anything, right?

Reply:
Replaced with getButtonTranslationKey. Done.